### DISCUSSION TITLE
react-query doesn't work properly when testing from different device

### DISCUSSION THREAD

**Discussion by matrs** [TanStack/query#4866] [Q&A] [answered]
Hello,
I'm developing a web site with `nextjs 12` and `react-query 4.22.0`. I have a page which uses react-query with `GetServerSideProps` and `dehydrate`,  an another page which doesn't use `react-query` at all.  When I serve the site with `npm run dev ` (`next dev`, the default), all the pages work as expected from my dev machine, but when I access the site from a mobile phone (using `{local-ip-address}:{port}/{page}`), all the pages render as expected,  but all the functionality related to `react-query` doesn't work and eventually fails with  `Error: Failed to fetch` (although I can see the proper response in the `react-query devtools`).  All the functionality which doesn't depend on `react-query` works as expected.  
I'm am missing something related to "`nextjs` / `react-query` development" here?

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

I think some more information here would be useful. When you say it doesn't work, what exactly do you mean? 

It sounds like you might be describing an erroneous Fetch request, something TanStack Query isn't responsible for.

**Reply by matrs** [reply]
Hello,
I have two pages. `index` and `/search`, where each make requests to a local database.  Only `index` uses react-query. Both work as expected in the dev machine, but only `/search` works  in my  "external" device. My index has a fetch to  a local database using `GetServerSideProps` , which works in my phone (page renders properly and i see the data in react-query devtools), but once I try to make another fetch from the index, it fails after a few seconds. These other requests come from a pagination component, where `pageN` and `selected` are the dependencies:

```js
const { data, isLoading, error, isFetching } = useQuery(
    ["products", { pageN: statePage, prodsPerPage: parseInt(selected) }],
    getApiProds,
    {
      keepPreviousData: true,
      refetchOnMount: false,
      refetchOnWindowFocus: false,
...
```

```js
....
// getApiProds
 try {
    response = await fetch(
      `http://localhost:3000/api/products?page=${pageN}&per_page=${prodsPerPage}`
    );
  } catch (error) {
    if (error instanceof Error) {
      ....
```

The confusing part for me is that everything works in my dev machine. 

On the other hand,  the `/search` page makes requests to the same database, that page uses `GetServerSideProps` but doesn't use any react-query, and that page works perfectly in my phone. **So my fetches are failing (they return `null`), but only when made within `useQuery` and from an external device**.

**Reply by louis-young** [reply]
So the issue is with the actual network request itself? If so, I doubt this is anything to do with TanStack Query. I'd suggest trying to debug this using the device or an emulator.

**Reply by matrs** [reply]
I just found the (basic) error. Maybe some people without that much experience in web can find this.
  My fetches were directed to `"localhost"` instead of  `{local-ip-address}`, so  they failed from an external device but not from the dev machine. The initial fetches worked regardless, because those are made directly to the server using server side props, without using my local API (as the documentation says), so no `localhost` or `{ip-address}` needed. 

Thanks you for your help

**Reply by louis-young** [reply]
Awesome, nice one 🚀 you’re welcome!
